src/main/java/com/training/springpractice/services/MovieService.java

```diff
@@ -0,0 +1,42 @@
+package com.training.springpractice.services;
+
+import com.training.springpractice.models.Actor;
+import com.training.springpractice.models.Movie;
+import com.training.springpractice.models.MovieDerived;
+import com.training.springpractice.repositories.ActorRepository;
+import com.training.springpractice.repositories.MovieDerivedRepository;
+import com.training.springpractice.repositories.MovieRepository;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+
+import java.util.List;
+import java.util.stream.Collectors;
+
+@Service("movieService")
+public class MovieService {
+
+    @Autowired
+    private MovieDerivedRepository movieDerivedRepository;
+
+    @Autowired
+    private ActorRepository actorRepository;
+
+    @Autowired
+    private MovieRepository movieRepository;
+
+    public MovieService() {}
+
+    public List<MovieDerived> findAll() {
+        return movieDerivedRepository.findByDeleted(false);
+    }
+
+    public Movie create (Movie movie) {
```

Comment:
According to the logic of your method what will happen in the following scenarios:

1. - The movie parameter is null
2. - The actors list in the movie parameter is empty.
3. - Some actor Ids does not exist in the db.
4. - The movieId does not exist in the db.